OrbitCore/LinuxUtils.cpp

```diff
@@ -54,46 +54,46 @@ namespace LinuxUtils {
 using ::ElfUtils::ElfFile;
 
 //-----------------------------------------------------------------------------
-std::vector<std::string> ReadProcMaps(pid_t pid) {
-  std::vector<std::string> modules;
-  // TODO: we should read the file directly instead or memory map it.
-  std::string result =
-      ExecuteCommand(absl::StrFormat("cat /proc/%d/maps", pid).c_str());
-
-  std::stringstream ss(result);
-  std::string line;
-  while (std::getline(ss, line, '\n')) {
-    modules.push_back(line);
-  }
-
-  return modules;
+outcome::result<std::vector<std::string>> ReadProcMaps(pid_t pid) {
+  std::filesystem::path maps_path{absl::StrFormat("/proc/%d/maps", pid)};
+  OUTCOME_TRY(maps_string, OrbitUtils::FileToString(maps_path));
+  return absl::StrSplit(maps_string, "\n");
 }
 
 //-----------------------------------------------------------------------------
-std::string ExecuteCommand(const char* a_Cmd) {
+outcome::result<std::string> ExecuteCommand(const std::string& cmd) {
   std::array<char, 128> buffer;
   std::string result;
-  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(a_Cmd, "r"), pclose);
+  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"),
+                                                pclose);
   if (!pipe) {
-    LOG("Could not open pipe");
+    ERROR("Failed to execute command \"%s\", error: %s (errno %d)", cmd.c_str(),
+          strerror(errno), errno);
```

Comment:
this should be the error message here. Also use SafeStrerror - because default libc strerror is not thread-safe.